src/rewrite/vct/rewrite/EncodeRangedFor.scala

```diff
@@ -0,0 +1,47 @@
+package vct.col.rewrite
+
+import hre.util.ScopedStack
+import vct.col.ast.{Select, Block, Assign, Expr, IterVariable, Local, LocalDecl, Loop, LoopInvariant, IterationContract, Eval, LoopContract, PostAssignExpression, Range, RangedFor, SeqMember, Statement, Variable}
+import vct.col.origin.AssignLocalOk
+import vct.col.util.AstBuildHelpers.const
+import vct.col.util.AstBuildHelpers._
+import vct.col.ast.RewriteHelpers._
+
+case object EncodeRangedFor extends RewriterBuilder {
+  override def key: String = "encodeRangedFor"
+  override def desc: String = "Encodes ranged for as a regular for loop"
+}
+
+case class EncodeRangedFor[Pre <: Generation]() extends Rewriter[Pre] {
+  val bounds = ScopedStack[(RangedFor[Pre], Expr[Post])]()
+
+  override def dispatch(stat: Statement[Pre]): Statement[Post] = stat match {
+    case rf @ RangedFor(iv @ IterVariable(iVar, from, to), contract, body) =>
+      implicit val o = iVar.o
+      val i = Local[Post](anySucc[Variable[Post]](iVar))(iVar.o)
```

Comment:
it should be `succ`, not `anySucc`. `anySucc` is only for references whose type is broader than any declaration kind, e.g. `Result(_: Ref[ContractApplicable])`.